addons/JobChange/jobchange.lua

```diff
@@ -0,0 +1,145 @@
+--[[
+Copyright © 2017, Sammeh of Quetzalcoatl
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of JobChange nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL Sammeh BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+]]
+
+_addon.name = 'Job Change'
+_addon.author = 'Sammeh'
+_addon.version = '1.0.3'
+_addon.command = 'jc'
+
+-- 1.0.1 first release
+-- 1.0.2 added 'reset' command to simply reset to existing job.  Changes sub job to a random starting job and back.
+-- 1.0.3 Code clean-up
+
+require('tables')
+packets = require('packets')
+res = require ('resources')
+
+function jobchange(job,main_sub)
+    windower.add_to_chat(4,"JobChange: Changing "..main_sub.." job to:"..res.jobs[job].ens)
+    if job and main_sub then 
+        local packet = packets.new('outgoing', 0x100, {
+            [main_sub:sub(1,1):upper()..main_sub:sub(2)..' Job'] = job,
+        })
+        packets.inject(packet)
+        coroutine.sleep(0.5)
+    end    
+end
+
+windower.register_event('addon command', function(command, ...)
+    local self = windower.ffxi.get_player()
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
```

Comment:
Here and below you call `:lower()` on the job, only to call `:upper()` on it later for the string comparison. This is very nitpicky at this point but I'd recommend just doing one, either lowercase or uppercase, and then sticking with it for the remainder of the addon. Since both *LuaCore* and *FFXI* itself use only uppercase to refer to job abbreviations, I'd use the same as well. Then you only need one `:upper()` call right here, and you can remove the `:lower()` call on line 68 and the two `:upper()` calls inside the `find_conflict` function.

Like I said, rather nitpicky at this point :) Just a recommendation to reduce unnecessary code where possible.